sdk/identity/src/token_credentials/client_secret_credentials.rs

```diff
@@ -7,6 +7,57 @@ use oauth2::{
 use std::{str, time::Duration};
 use url::Url;
 
+/// Provides options to configure how the Identity library makes authentication
+/// requests to Azure Active Directory.
+#[derive(Clone, Debug, PartialEq)]
+pub struct TokenCredentialOptions {
+    authority_host: String,
+}
+
+impl Default for TokenCredentialOptions {
+    fn default() -> Self {
+        Self {
+            authority_host: authority_hosts::AZURE_PUBLIC_CLOUD.to_owned(),
+        }
+    }
+}
+
+impl TokenCredentialOptions {
+    pub fn new(authority_host: &str) -> Self {
```

Comment:
In the case where you're just going to call `to_owned` on the provided string no matter what, it's best to take the argument as a `String`.

Reply:
I'd also add some documentation  highlighting that users may want to use the `Default` implementation.

Reply:
done